### PR TITLE
add alpha channel support fo png files

### PR DESCRIPTION
**Pull request by mohamedali-zorgati1997**

### CODE REVIEW

**Comment by MartinThoma** [comment]
Do you happen to have an example PDF which shows the difference? What does it mean that alpha channel is now supported?

**Comment by mohamedali-zorgati1997** [comment]
I just created [this](https://drive.google.com/file/d/12ulyJkXGdBCe92_uMY5c8TThxfxDNOM5/view?usp=sharing) PDF file and it is a perfect example. Try to extract the image inside with and without the change I made and you will see the difference.
The alpha channel is the transparency mask in PNG files, so when extracting PNG images without the alpha channel, the transparent areas will show in black.

File: Scripts/pdf-image-extractor.py [hunk 1]
```diff
@@ -34,6 +34,9 @@
                 if '/Filter' in xObject[obj]:
                     if xObject[obj]['/Filter'] == '/FlateDecode':
                         img = Image.frombytes(mode, size, data)
+                        if "/SMask" in XObject[obj]: # add alpha channel
+                            alpha = Image.frombytes("L", size, XObject[obj]["/SMask"].getData())
```
Comment (MartinThoma) [inline Scripts/pdf-image-extractor.py:38]:
```suggestion
                        if "/SMask" in xObject[obj]: # add alpha channel
                            alpha = Image.frombytes("L", size, xObject[obj]["/SMask"].getData())
```

**Comment by MartinThoma** [comment]
Nice! Thank you for your contribution!

There was a tiny typo which I fixed - besides that, it looks good!

**Comment by MartinThoma** [comment]
Thank you very much that you took the time to respond, although it took a year until you've got a reaction :hugs:

**Comment by mohamedali-zorgati1997** [comment]
Always happy to be of service.

**Pull request merged**